projects/laji/src/app/+project-form/form/document-form/document-form.facade.ts

```diff
@@ -0,0 +1,485 @@
+import { NamedPlace } from '../../../shared/model/NamedPlace';
+import { TemplateForm } from '../../../shared-modules/own-submissions/models/template-form';
+import { FooterService } from '../../../shared/service/footer.service';
+import { Document } from '../../../shared/model/Document';
+import { TranslateService } from '@ngx-translate/core';
+import { LatestDocumentsFacade } from '../../../shared-modules/latest-documents/latest-documents.facade';
+import { DocumentService } from '../../../shared-modules/own-submissions/service/document.service';
+import { Form } from '../../../shared/model/Form';
+import { FormService } from '../../../shared/service/form.service';
+import { Util } from '../../../shared/service/util.service';
+import { UserService } from '../../../shared/service/user.service';
+import { FormPermissionService, Rights } from '../../../shared/service/form-permission.service';
+import { NamedPlacesService } from '../../../shared/service/named-places.service';
+import { DocumentStorage } from '../../../storage/document.storage';
+import { DocumentApi } from '../../../shared/api/DocumentApi';
+import * as deepmerge from 'deepmerge';
+import * as moment from 'moment';
+import { LocalStorage } from 'ngx-webstorage';
+import { Global } from 'projects/laji/src/environments/global';
+import { PersonApi } from '../../../shared/api/PersonApi';
+import { Person } from '../../../shared/model/Person';
+import { Annotation } from '../../../shared/model/Annotation';
+import { LajiApi, LajiApiService } from '../../../shared/service/laji-api.service';
+import { Logger } from '../../../shared/logger';
+import { LajiFormUtil } from '@laji-form/laji-form-util.service';
+
+export enum FormError {
+  notFoundForm = 'notFoundForm',
+  notFoundDocument = 'notFoundDocument',
+  loadFailed = 'loadFailed',
+  noAccess = 'noAccess',
+  noAccessToDocument = 'noAccessToDocument',
+  templateDisallowed = 'templateDisallowed',
+  missingNamedPlace = 'missingNamedPlace'
+}
+
+export function isFormError(any: any): any is FormError {
+  return typeof any === 'string' && !!FormError[any];
+}
+
+export enum Readonly {
+  noEdit,
+  false,
+  true
+}
+
+export interface SaneInputModel {
+  form: Form.SchemaForm;
+  formData: any;
+  hasChanges: boolean;
+  template: boolean;
+  namedPlace?: NamedPlace;
+}
+
+export type InputModel = SaneInputModel | FormError;
+
+export interface SaneViewModel extends SaneInputModel {
+  readonly: Readonly;
+  isAdmin: boolean;
+  namedPlaceHeader: string[];
+  locked?: boolean;
+  editingOldWarning?: string;
+}
+
+export function isSaneViewModel(any: ViewModel): any is SaneViewModel {
+  return typeof any !== 'string' && any && !!any?.form;
+}
+
+export type ViewModel = SaneViewModel | FormError;
+
+interface DocumentAndHasChanges {
+  document: Document;
+  hasChanges: boolean;
+}
+
+@Injectable()
+export class DocumentFormFacade {
+  @Input() formID: string;
```

Comment:
There shouldn't be any inputs here and it seems that these variables are not used

Reply:
Thanks, fixed that